File: vite-front-end/src/entities/Flight/FlightData.ts
import {Aircraft} from "./Aircraft.ts";
import {Departure} from "./Departure.ts";
import {Arrival} from "./Arrival.ts";
import {Flight} from "./Flight.ts";

export type FlightData = {
    flight: Flight
    arrival: Arrival
    departure: Departure
    aircraft: Aircraft
}

File: Flight API/src/main/java/me/abdul/flightapi/entities/FlightsResponseEntity/FlightData.java
package me.abdul.flightapi.entities.FlightsResponseEntity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class FlightData {
    public Flight flight;
    public Arrival arrival;
    public Departure departure;
    public Aircraft aircraft;

}
